spec/features/audit/report/csv_export_spec.rb

```diff
@@ -13,99 +7,145 @@ def content_disposition
     page.response_headers.fetch("Content-Disposition")
   end
 
-  context "Two content items are in the database" do
-    let!(:hmrc) {
-      create(
-        :content_item,
-        title: "HMRC",
-        document_type: "organisation"
-      )
-    }
-
-    before do
-      example1 = create(
-        :content_item,
-        title: "Example 1",
-        base_path: "/example1",
-        allocated_to: me,
-      )
-
-      create(
-        :link,
-        source_content_id: example1.content_id,
-        target_content_id: hmrc.content_id,
-        link_type: "primary_publishing_organisation",
-      )
-
-      create(
-        :content_item,
-        title: "Example 2",
-        base_path: "/example2",
-      )
-
-      create(:audit, content_item: example1)
-    end
-
-    scenario "Exporting a csv file as an attachment" do
-      visit audits_report_path
-      click_link "Export filtered audit to CSV"
-
-      expect(content_type).to eq("text/csv")
-      expect(content_disposition).to start_with("attachment")
-      expect(content_disposition).to include(
-        'filename="Transformation_audit_report_CSV_download.csv"',
-      )
+  scenario "Exporting a csv file as an attachment" do
+    given_i_am_exporting_an_audit_report
+    then_i_receive_an_audit_progress_report_in_csv_format
+  end
 
-      expect(page).to have_content("Title,URL")
-      expect(page).to have_content("Example 1,https://gov.uk/example1")
-    end
+  scenario "Applying the filters to the export" do
+    given_i_have_applied_filters_and_exported_audits
+    then_i_receive_an_audit_progress_report_for_filtered_audits_in_csv_format
+  end
 
-    scenario "Applying the filters to the export" do
-      visit audits_report_path
-      select "Anyone", from: "allocated_to"
+  scenario "Multiple pagesincluding_details_of_audit_progress_ of content items are in the database" do
+    given_i_have_multiple_pages_of_content_items_and_do_not_filter_them
+    then_i_receive_an_audit_progress_report_for_all_audits_in_csv_format
+  end
 
-      select "HMRC", from: "Organisations"
-      click_on "Apply filters"
+  scenario "Discard audit status filter when clicking from content view to report, and then exporting CSV" do
+    given_i_apply_filters_to_the_audit_content_page
+    then_i_see_filtered_audits
+    given_that_i_navigate_to_audits_report_page
+    then_i_see_an_unfiltered_audit_progress_report
+  end
 
-      click_link "Export filtered audit to CSV"
-      expect(page).to have_content("Example 1,https://gov.uk/example1")
-      expect(page).to have_no_content("Example 2,https://gov.uk/example2")
-    end
+  def given_i_am_exporting_an_audit_report
```

Comment:
My above comments also apply to this setup step - there's too much happening in here that isn't hinted at by the name of the method.